cmd/ocm/gcp/update-wif-config.go

```diff
@@ -1,39 +1,106 @@
 package gcp
 
 import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/openshift-online/ocm-cli/pkg/gcp"
+	"github.com/openshift-online/ocm-cli/pkg/ocm"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 var UpdateWifConfigOpts struct {
-	wifId      string
-	templateId string
 }
 
 // NewUpdateWorkloadIdentityConfiguration provides the "gcp update wif-config" subcommand
 func NewUpdateWorkloadIdentityConfiguration() *cobra.Command {
 	updateWifConfigCmd := &cobra.Command{
-		Use:     "wif-config",
+		Use:     "wif-config [ID|Name]",
 		Short:   "Update wif-config.",
 		RunE:    updateWorkloadIdentityConfigurationCmd,
 		PreRunE: validationForUpdateWorkloadIdentityConfigurationCmd,
 	}
 
-	updateWifConfigCmd.PersistentFlags().StringVar(&UpdateWifConfigOpts.wifId, "wif-id", "",
-		"Workload Identity Federation ID")
-	updateWifConfigCmd.MarkPersistentFlagRequired("wif-id")
-	updateWifConfigCmd.PersistentFlags().StringVar(&UpdateWifConfigOpts.templateId, "template-id", "",
-		"Template ID")
-	updateWifConfigCmd.MarkPersistentFlagRequired("template-id")
-
 	return updateWifConfigCmd
 }
 
 func validationForUpdateWorkloadIdentityConfigurationCmd(cmd *cobra.Command, argv []string) error {
-	// No validation needed
+	if err := wifKeyArgCheck(argv); err != nil {
+		return err
+	}
 	return nil
 }
 
 func updateWorkloadIdentityConfigurationCmd(cmd *cobra.Command, argv []string) error {
-	// No implementation yet
+	ctx := context.Background()
+	log := log.Default()
+	key := argv[0]
```

Comment:
I have readability concerns with the way we are deriving the key throughout our wif commands. 

We know that the first element of argv exists and will be populated because we run wifKeyArgCheck on argv within the preRunE function that is run prior to RunE; we are safe to access argv[0]. However knowing this requires the reader to be aware of this execution flow, which is defined by the cobra package.

I'm thinking it may be better for the readibility of the code to do away with the preRunE call, and do the flag validation at the start of RunE.  We could then have functions that both validate that the data is there and return it:
```
key, err := wifKeyFromArgs(argv)
```

Reply:
I moved to a helper function and took it out of the preRun validation